src/reader/ft_split_smart.c

```diff
@@ -0,0 +1,120 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split_smart.c                                   :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: al7aro <marvin@42.fr>                      +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2022/09/14 11:32:13 by al7aro            #+#    #+#             */
+/*   Updated: 2022/09/19 12:08:09 by al7aro-g         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#include "libft.h"
+#include "tab.h"
+
+int	is_special(char *str)
+{
+	int	i;
+
+	i = 0;
+	if (*str == '<')
+	{
+		while (*(str + i) == '<')
+			i++;
+		return (i);
+	}
+	if (*str == '>')
+	{
+		while (*(str + i) == '>')
+			i++;
+		return (i);
+	}
+	if (*str == '|')
+		return (1);
+	if (*str == ';')
+		return (1);
+	return (0);
+}
+
+int	is_word(char *str)
+{
+	char	del;
+	int		i;
+
+	i = -1;
+	if (is_special(str))
+		return (is_special(str));
+	del = ' ';
+	while (*(str + ++i) && *(str + i) != del)
+	{
+		if (del == ' ' && is_special(str + i))
+			return (i);
+		if ((*(str + i) == '\"' && *(str + i) != del && del != '\'')
+			|| (*(str + i) == '\'' && *(str + i) != del && del != '\"'))
+			del = *(str + i);
+		if (del != ' ' && *(str + i + 1) == del)
+		{
+			del = ' ';
+			i += 1;
+		}
+	}
+	return (i + 1 + (del == '\"' || del == '\'') - (*(str + i) == ' '));
+}
+
+int	cnt_words(char *str)
```

Comment:
since there are many functions related to "words" i think you can just create a new class called "word_util"